### ISSUE TITLE
MongoDB - Field Filter - not including 31st July

### ISSUE THREAD

**Issue by manjeshpv** [metabase/metabase#11859]
**Describe the bug**

Normal Query:

```json
 [{ "$match": { "_created_at": { "$gte": ISODate("2019-07-01T00:00:00.000Z"), "$lte": ISODate("2019-07-31T23:59:59.999Z") } } }]
```

same query with field filter
```json
 { "$match": {{created_at}}  },
```

**To Reproduce**
1. Click on ASK QUESTION, Normal Native query, MongoDB Database should be there
2. Keep some records created_at is 31st July
3. apply field filter
4. In results, you will not find records for 31st july

**Expected behavior**

- field filter should include 1st of the month(2019-07-01 00:00:00 to 2019-07-31 23:59:59 )

**Information about your Metabase Installation:**
Version v0.34.1
Built on 2020-01-13
```json
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.5+10-post-Ubuntu-0ubuntu1.118.04",
    "java.vendor": "Private Build",
    "java.vendor.url": "Unknown",
    "java.version": "11.0.5",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.5+10-post-Ubuntu-0ubuntu1.118.04",
    "os.name": "Linux",
    "os.version": "4.18.0-1013-azure",
    "user.language": "en",
    "user.timezone": "Asia/Kolkata"
  },
  "metabase-info": {
    "databases": [
      "mongo"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "date": "2020-01-13",
      "tag": "v0.34.1",
      "branch": "release-0.34.x",
      "hash": "265695c"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

![image](https://user-images.githubusercontent.com/1999957/73834497-d9080400-4831-11ea-89ab-75b4c2d2f94a.png)

**Severity**
blocking

because I was waiting for field filters from a year, Now it is available, without this reports become inconsistent, I feel all mongodb users may got impacted by this

https://github.com/metabase/metabase/issues/11690

**Comment by arjun3396** [comment]
I am also facing the same issue

**Comment by dacort** [comment]
Hi @manjeshpv - this is fixed on [v0.34.2](https://github.com/metabase/metabase/releases/tag/v0.34.2), released today. Please upgrade and that should fix the issue, thanks!

Duplicate of #11715
